### PR TITLE
[SPARK-29685][SQL] Adds format to the spark-sql output like its implemented by Dataset showString

### PR DESCRIPTION
**Pull request by javierivanov**
### What changes were proposed in this pull request?

This PR adds a formatted print method like it is provided in the showString implementation from Dataset while using the spark-sql client.

Actual Spark-sql client
```
spark-sql> select * from table1;
5       name3   add3
5       name1   add1
5       name2   add2
```

Spark Dataset show
```
scala> sql("select * from table1").show()
+---+-----+-------+
| id| name|address|
+---+-----+-------+
|  5|name3|   add3|
|  5|name1|   add1|
|  5|name2|   add2|
+---+-----+-------+
```
### Why are the changes needed?

To have a consistent output between Dataset show and spark-sql interfaces.

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

org.apache.spark.sql.hive.thriftserver.CliSuite.scala was adapted for the new fortmat. Code compiles and pass tests.

### CODE REVIEW

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Thank you for making a PR, @javierivanov .
`GitHub Action` seems to complain about the Scala coding style.
Could you run `dev/scalastyle` and fix all of them?

**Comment by javierivanov** [comment]
Thanks for the quick response @dongjoon-hyun. All good now.

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala [hunk 1]
```diff
@@ -387,9 +388,9 @@ class CliSuite extends SparkFunSuite with BeforeAndAfterAll with Logging {
 
   test("SPARK-26321 Should not split semicolon within quoted string literals") {
     runCliWithin(3.minute)(
-      """select 'Test1', "^;^";""" -> "Test1\t^;^",
-      """select 'Test2', "\";";""" -> "Test2\t\";",
-      """select 'Test3', "\';";""" -> "Test3\t';",
+      """select 'Test1', "^;^";""" -> "Test1|^;^",
+      """select 'Test2', "\";";""" -> "Test2| \";",
+      """select 'Test3', "\';";""" -> "Test3| ';",
```
Comment (HyukjinKwon) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala]: I think the output was matched to Hive's intentionally.

Comment (javierivanov) [reply sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala]: Do you mean the extra space? the minimun col width is 3

**Comment by wangyum** [comment]
Hi @javierivanov Are you interested in implementing a new Spark CLI similar to `spark-sql`, but does not include any Hive dependency.

**Comment by javierivanov** [comment]
Yeah, sure. The idea is to have a pure Scala implementation ?

**Comment by javierivanov** [comment]
@wangyum do you mean a different module or related to this PR #26340?

**Pull request closed without merging**